cordapp-example/contracts-kotlin/src/main/kotlin/com/example/contract/CashContract.kt

```diff
@@ -0,0 +1,60 @@
+package com.example.contract
+
+import com.example.state.CashState
+import net.corda.core.contracts.CommandData
+import net.corda.core.contracts.Contract
+import net.corda.core.contracts.requireSingleCommand
+import net.corda.core.contracts.requireThat
+import net.corda.core.transactions.LedgerTransaction
+
+class CashContract : Contract {
+    companion object {
+        @JvmStatic
+        val ID = CashContract::class.java.canonicalName
+    }
+
+    override fun verify(tx: LedgerTransaction) {
+        val command = tx.commands.requireSingleCommand<Commands>()
```

Comment:
I added option to create many `CashState`s at once. `CreateMoneyFlowTest` passes. But I have issue when testing payback flow. In this test transaction has 3 commands: `Destroy` (for IUO) and `Move`(s) (for each cash). Problem is that in line above `tx.commands` returns all three of them:

![image](https://user-images.githubusercontent.com/34737849/79494651-a9d8f480-8023-11ea-9b1c-14235f5f1eed.png)

and because of that, above line fails with message `List has more than one element.`.
When I'm building tx I'm adding two `Move` commands for every state:
```
cashStatesAndRefs.iterator().forEach {
                val (moveCashCommand, cashOutputState) = it.state.data.withNewOwner(iouState.lender)
                txBuilder.addInputState(it)
                txBuilder.addOutputState(cashOutputState, it.state.contract, it.state.notary)
                txBuilder.addCommand(moveCashCommand, it.state.data.owner.owningKey)
            }
```
and this is probably the issue. But how can I do this differently, if I need to call `withNewOwner` on every state and then I got output and move values that (probably) I need to include in tx?

Reply:
done. commit is comming...